test/token_whitelist/add_methodIDs_test.go

```diff
@@ -0,0 +1,141 @@
+package token_whitelist_test
+
+import (
+    "github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/tokencard/contracts/v2/test/shared"
+	"github.com/tokencard/ethertest"
+)
+
+var _ = Describe("method Ids", func() {
+
+    When("the tokenWhitelist is deployed", func() {
+        It("it should have 3 methods (transfer, approve, transferFrom) whitelisted", func() {
+
+            methodID := crypto.Keccak256Hash([]byte("transfer(address,uint256)"))
+            var method [4]byte
+            copy(method[:], methodID[:4])
+            wl, err := TokenWhitelist.IsERC20MethodWhitelisted(nil, method)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(wl).To(BeTrue())
+
+            methodID = crypto.Keccak256Hash([]byte("approve(address,uint256)"))
+            copy(method[:], methodID[:4])
+            wl, err = TokenWhitelist.IsERC20MethodWhitelisted(nil, method)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(wl).To(BeTrue())
+
+            methodID = crypto.Keccak256Hash([]byte("transferFrom(address,address,uint256)"))
+            copy(method[:], methodID[:4])
+            wl, err = TokenWhitelist.IsERC20MethodWhitelisted(nil, method)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(wl).To(BeTrue())
+        })
+    })
+
+    When("the controller admin adds methods (one already existing)", func() {
+
+        var method1 [4]byte
+        var method2 [4]byte
+
+        BeforeEach(func() {
+            methodID := crypto.Keccak256Hash([]byte("transfer(address,uint256)"))
+            copy(method1[:], methodID[:4])
+
+            methodID = crypto.Keccak256Hash([]byte("increaseApproval(address,uint256)"))
+            copy(method2[:], methodID[:4])
+
+            methods := [][4]byte{method1, method2}
+            tx, err := TokenWhitelist.AddMethodIds(ControllerAdmin.TransactOpts(), methods)
+            Expect(err).ToNot(HaveOccurred())
+            Backend.Commit()
+            Expect(isSuccessful(tx)).To(BeTrue())
+        })
+
+        It("should add 1 new method", func() {
+            methodID := crypto.Keccak256Hash([]byte("increaseApproval(address,uint256)"))
+            copy(method1[:], methodID[:4])
```

Comment:
it's not really cool to set variable with a scope outside of `It` in the `It` function - we had testing bugs because of that before.